ffi/src/winscard/system_scard/mod.rs

```diff
@@ -0,0 +1,166 @@
+#![cfg(feature = "scard")]
+
+#[macro_use]
+mod macros;
+
+mod card;
+mod context;
+
+use std::borrow::Cow;
+
+pub use card::SystemScard;
+pub use context::SystemScardContext;
+#[cfg(target_os = "windows")]
+use ffi_types::winscard::functions::SCardApiFunctionTable;
+use winscard::WinScardResult;
+
+fn parse_multi_string(buf: &[u8]) -> WinScardResult<Vec<&str>> {
+    let res: Result<Vec<&str>, _> = buf
+        .split(|&c| c == 0)
+        .filter(|v| !v.is_empty())
+        .map(std::str::from_utf8)
+        .collect();
+
+    Ok(res?)
+}
+
+fn parse_multi_string_owned(buf: &[u8]) -> WinScardResult<Vec<Cow<'static, str>>> {
+    Ok(parse_multi_string(buf)?
+        .into_iter()
+        .map(|r| Cow::Owned(r.to_owned()))
+        .collect())
+}
```

Comment:
suggestion: If always returning an owned string, use `String` instead of `Cow<'a, str>`. `Cow` is used for types which may be owned or not. (`Cow<'static, str>` is a special case for types which are `'static` even if the value borrowed.)

Reply:
Yes, I agree with you. I have chosen the `Cow` instead of `String` because later the returned value will be transformed to the owned `Cow` anyway (because of the `WinSCardContext` trait API)